librz/include/rz_arch.h

```diff
@@ -59,6 +61,245 @@ RZ_API bool rz_arch_load_platform_sdb(RZ_NONNULL RzArchPlatformTarget *t, RZ_NON
 RZ_API bool rz_arch_platform_init(RzArchPlatformTarget *t, RZ_NONNULL const char *arch, RZ_NONNULL const char *cpu,
 	const char *platform, RZ_NONNULL const char *platforms_dir);
 
+typedef enum {
+	RZ_ARCH_INFO_CODE_ALIGN = 0, ///< code/text segment alignment
+	RZ_ARCH_INFO_DATA_ALIGN, ///< data segment alignment
+	RZ_ARCH_INFO_INSTRUCTION_MIN_SIZE, ///< minimum instruction length
+	RZ_ARCH_INFO_INSTRUCTION_MAX_SIZE, ///< maximum instruction length
+	RZ_ARCH_INFO_ADDRESS_SPACE, ///< address space bit-size
+	/* ignore */
+	RZ_ARCH_INFO_END, ///< Used only to define the max value of RZ_ARCH_INFO_* enum
+} RzArchInfo;
+
+typedef enum {
+	RZ_ARCH_EXTENSION_TYPE_GENERIC = 0, ///< Opcode extension generic
+	RZ_ARCH_EXTENSION_TYPE_FLOATING_POINT, ///< Opcode extension floating point
+	RZ_ARCH_EXTENSION_TYPE_VECTORIAL, ///< Opcode extension vectorial
+	RZ_ARCH_EXTENSION_TYPE_VIRTUALIZATION, ///< Opcode extension virtualization
+	RZ_ARCH_EXTENSION_TYPE_PRIVILEGED, ///< Opcode extension for instructions with privilieged execution
+	RZ_ARCH_EXTENSION_TYPE_MMX, ///< Opcode extension for multimedia instructions
```

Comment:
Isn't better to group them in something like:
- SIMD
- Vector extensions
- AUTH
?
It will be more generic.